airflow/datasets/metadata.py

```diff
@@ -33,7 +33,11 @@ class Metadata:
 
     uri: str
     extra: dict[str, Any]
+    alias: DatasetAlias | str | None
```

Comment:
Should this just be `DatasetAlias | None` and we convert str eagerly?

Reply:
yep, sound good. just updated.